packages/components/src/components/input/input.stories.ts

```diff
@@ -0,0 +1,642 @@
+  // Get input element and buttons
+  const inputEl = document.querySelector('sd-input')!;
+  const stepDownButton = document.querySelector('#stepDownButton')!;
+
+  // function to add class if not present
+  const addClass = (el: Element, className: string) => {
+    if (!el.classList.contains(className)) {
+      el.classList.add(className);
+    }
+  };
+
+  // function to remove class if present
+  const removeClass = (el: Element, className: string) => {
+    if (el.classList.contains(className)) {
+      el.classList.remove(className);
+    }
+  };
+
+  // function to adjust classlist of stepDownButton
+  const adjustClasslist = () => {
+    if (inputEl.value === '0') {
+      addClass(stepDownButton, 'text-neutral-500');
+    } else {
+      removeClass(stepDownButton, 'text-neutral-500');
+    }
+  };
+
+  // Call sd-input stepUp() or stepDown() method
+  if (direction === 'up') inputEl.stepUp();
+  if (direction === 'down') inputEl.stepDown();
+
+  // Adjust input value to 2 decimals (currency)
+  inputEl.value = String(parseInt(inputEl.value, 10).toFixed(2));
+
+  // Invoke adjustClasslist()
+  adjustClasslist();
+};
+
+/**
+ * Sample implementation of a currency stepper.
+ */
+
+export const Samples = {
+  parameters: {
+    controls: {
+      exclude: [
+        'title',
+        'type',
+        'size',
+        'inputmode',
+        'value',
+        'placeholder',
+        'label',
+        'help-text',
+        'message',
+        'clearable',
+        'disabled',
+        'readonly',
+        'password-toggle',
+        'password-visible',
+        'no-spin-buttons',
+        'minlength',
+        'maxlength',
+        'min',
+        'max',
+        'form',
+        'name',
+        'required',
+        'pattern',
+        'step'
+      ]
+    }
+  },
+  render: () => {
+    return html`
+      <style>
+        .headline {
+          margin-bottom: 16px;
+          padding: 16px;
+          background: #e0e0e0;
+          text-align: left;
+          font-size: 14px;
+          font-weight: bold;
+          width: 100%;
+          box-sizing: border-box;
+        }
+      </style>
+      <div>
+        <div class="headline">sd-input with currency stepper</div>
+        <sd-input type="number" min="0" class="w-[231px]"
+          ><span slot="right" class="text-sm inline-flex items-center"
+            ><span class="text-neutral-700">EUR</span>
+            <button
+              id="stepDownButton"
+              @click=${() => step('down')}
+              class="ml-4 scale-[1.714] inline-flex items-center text-neutral-500"
+            >
+              <sd-icon library="global-resources" name="system/minus-round"></sd-icon>
+            </button>
+            <button id="stepUpButton" @click=${() => step('up')} class="ml-4 scale-[1.714] inline-flex items-center">
+              <sd-icon library="global-resources" name="system/plus-round"></sd-icon></button
+          ></span>
+        </sd-input>
```

Comment:
```suggestion
<sd-input id="numericInput" type="number" min="0" class="w-[231px]">
  <span slot="right" class="text-sm inline-flex items-center gap-4">
    <span class="text-neutral-700">EUR</span>
    <button
      onclick="document.getElementById('numericInput').stepDown()"
      class="scale-[1.714] inline-flex items-center text-neutral-500"
    >
      <sd-icon library="global-resources" name="system/minus-round"></sd-icon>
    </button>
    <button
      onclick="document.getElementById('numericInput').stepUp()"
      class="scale-[1.714] inline-flex items-center"
    >
      <sd-icon library="global-resources" name="system/plus-round"></sd-icon>
    </button>
  </span>
</sd-input>
```

Could you try something like this?

Reply:
That is what I originally did, it will not apply conditional styles or currency format which reduces the quality of the story.  

@yoezlem clicking the step buttons fires my step function which I took the time to implement for the sake of this sample and furthermore documented via comments:
https://github.com/solid-design-system/solid/assets/39494579/34b6d104-11e1-4bd6-bc06-23a622ae33b9

Is it not working for you?  If not I would need details to further understand.  We can either refine that logic or remove it and have a simple story with no formatting or conditional styles.  Either is fine for me.

Reply:
Ah, just saw now the method. It's super nice.

if you switch to docs where all stories are it's not working.

Does it work with onclick instead of @click? Or do we maybe have elements with the same id or similar?

Reply:
I just realized that the problem for the docs might be, that your function is maybe not available inside the story.


<img width="1036" alt="CleanShot 2023-11-09 at 08 24 25@2x" src="https://github.com/solid-design-system/solid/assets/26542182/be9dab78-70c5-41b7-af46-df0b65fa47c9">

That's why I wasn't really aware of what the method is doing. 
 So: Could you insert all this JS inside the story? This is especially important as people should be able to see, what you're doing and if you have it outside the story that doesn't work. :)

Reply:
Exactly! It works only within the documentation sample, not in the actual story. Strange behavior 🤔

Reply:
This is what I originally did, however then step is not recognized.
![image](https://github.com/solid-design-system/solid/assets/39494579/0b4e1c70-648e-4d56-8119-4b2d67e0bc45)

I tried several workarounds and published the result because it was working in my story.  I'm not sure how much more time that is worth.  It will be much easier to implement in an actual counter component.  

I'll try a couple more things.